bank/src/main.rs

```diff
@@ -45,29 +53,63 @@ fn transfer_funds(txn: &mut Transaction, from: i64, to: i64, amount: i64) -> Res
     Ok(())
 }
 
-fn ssl_config() -> Result<MakeTlsConnector, ErrorStack> {
-    let mut builder = SslConnector::builder(SslMethod::tls())?;
-    builder.set_ca_file("certs/ca.crt")?;
-    builder.set_certificate_chain_file("certs/client.maxroach.crt")?;
-    builder.set_private_key_file("certs/client.maxroach.key", SslFiletype::PEM)?;
-    Ok(MakeTlsConnector::new(builder.build()))
+fn delete_accounts(txn: &mut Transaction) -> Result<(), Error> {
+    txn.execute(
+        "DELETE FROM accounts", &[],
+    )?;
+    Ok(())
 }
 
 fn main() {
     let connector = ssl_config().unwrap();
+    let connection_uri = env!("DATABASE_URL");
     let mut client =
-        Client::connect("postgresql://maxroach@localhost:26257/bank", connector).unwrap();
+        Client::connect(&connection_uri, connector).unwrap();
 
+    // Create the "accounts" table.
+    client
+        .execute(
+            "CREATE TABLE IF NOT EXISTS accounts (id UUID PRIMARY KEY, balance INT)",
+            &[],
+        )
+        .unwrap();
+
+    let id1 = Uuid::new_v4();
+    let id2 = Uuid::new_v4();
+    // Insert two rows into the "accounts" table.
+    client
+        .execute(
+            "INSERT INTO accounts (id, balance) VALUES ($1, 1000), ($2, 250)",
+            &[&id1, &id2],
+        )
+        .unwrap();
+
+    // Print out the balances.
+    println!("Balances at {:?}:", Utc::now());
+    for row in &client
+        .query("SELECT id, balance FROM accounts", &[])
+        .unwrap()
+    {
+        let id: Uuid = row.get(0);
+        let balance: i64 = row.get(1);
+        println!("account id: {}  balance: {}", id, balance);
+    }
+    
     // Run a transfer in a transaction.
-    execute_txn(&mut client, |txn| transfer_funds(txn, 1, 2, 100)).unwrap();
+    execute_txn(&mut client, |txn| transfer_funds(txn, id1, id2, 100)).unwrap();
 
     // Check account balances after the transaction.
+    println!("Final balances at {:?}:", Utc::now());
     for row in &client
         .query("SELECT id, balance FROM accounts", &[])
         .unwrap()
     {
-        let id: i64 = row.get(0);
+        let id: Uuid = row.get(0);
         let balance: i64 = row.get(1);
-        println!("{} {}", id, balance);
+        println!("account id: {}  balance: {}", id, balance);
     }
+    
+    // Delete the accounts
+    execute_txn(&mut client, |txn| delete_accounts(txn)).unwrap();
```

Comment:
Sometimes I put this kind of cleanup step at the start so that it'll still work even if the script failed partway through on the previous run.

Reply:
Done.